New feature: Item progress bars in various menus

violine1101 opened the pull request:
This PR adds item progress bars to various menus (reusing the vanilla code for durability bars).

There are three types of these progress bars. First, a general progress bar (green) for all menu items that have some sort of percentage progress bar in their tooltip. This includes quests, fishing islands, badges, cosmetic groups, and a few more things.

<img width="2560" height="1440" alt="2025-08-23_15 01 00" src="https://github.com/user-attachments/assets/f07005da-2e89-4e47-879c-67be4bab4d5d" />
<img width="2560" height="1440" alt="2025-08-23_15 01 30" src="https://github.com/user-attachments/assets/09ceb7d1-5263-45b2-b10e-17833651710c" />
<img width="2560" height="1440" alt="2025-08-23_15 00 53" src="https://github.com/user-attachments/assets/8cf85290-4c74-453d-ad2a-8207cbd391cf" />

Next, cosmetics that have already been redeemed have a purple progress bar showing how often it has been donated to the royal collection. This was actually the main reason for why I wanted to implement this: I found it difficult to distinguish which cosmetics I already have and which ones I don't. Those which I already redeemed before now have such a bar, and those that are new don't.

<img width="2560" height="1440" alt="2025-08-23_02 43 31" src="https://github.com/user-attachments/assets/4699d132-8131-44c3-bc37-ac960670dd38" />
<img width="2560" height="1440" alt="2025-08-23_02 43 14" src="https://github.com/user-attachments/assets/9bfe3d7c-23ff-47d6-8f7c-7e13b67deaf8" />

And finally, there's also bars for cosmetics in the cosmetic collection menu, showing which chroma packs have been unlocked. The bar color changes based on which colors are available :)

<img width="2560" height="1440" alt="2025-08-23_13 15 16" src="https://github.com/user-attachments/assets/de4341d6-97e1-40d3-8bea-5f3ff9f576e8" />

Of course I also added separate config options for these three types of progress bars.

LukynkaCZE commented:
very cool stuff!

kispintyi commented:
The green color looks weird in this instance, its like they are broken or something which would definitely throw me up, apart from that cool! :D

violine1101 commented:
Thanks for the input! I've done the following changes:
* Changed the green progress bar to white by default
* Added a config option to choose a different color for the (regular) progress bars
* Progress bars will only be completely empty if the progress is absolute zero

Hopefully that solves your issue that the bars make these items look like they're "broken". Let me know what you think!

<img height="480" alt="2025-09-01_02 31 51 (cropped)" src="https://github.com/user-attachments/assets/7a83af18-565c-4f9c-a88c-55a927f75a5f" />

kispintyi commented:
Wow kinda cool, thanks! :D

AsoDesu commented:
very very cool stuff, 
i'm gonna have a quick look at it now, i've just merged #209 which has caused a few conflicts (sorry i think i should have merged this one first cuz it's older and bigger lol) 
but i do like where this PR is going :D
